src/kernel/src/ucred/mod.rs

```diff
@@ -125,3 +193,31 @@ impl Errno for PrivilegeError {
         }
     }
 }
+
+#[derive(Debug, Error)]
+pub enum PrisonCheckError {
+    #[error("Prison check failed")]
+    CheckFailed,
```

Comment:
The error is unclear what going on. The message also does not align with Rust guidance.